autopush/db.py

```diff
@@ -363,6 +379,7 @@ def fetch_notifications(self, uaid):
 
     @track_provisioned
     def save_notification(self, uaid, chid, version):
+        # type: (str, str, Optional[int]) -> bool
```

Comment:
Yep, this is simplepush legacy cruft. "Version" here is an identifier that prevented delete from accidentally deleting the wrong "version" of the notification (e.g. if a remote client got a message, then an update occurred while in process, the client wouldn't nuke the newer message.) Likewise if we got a notification "out of order" we wouldn't overwrite the newer notification with an older one. 

Technically, "version" isn't optional, at least in the older code. It's an int that is coverted to a string because that's a handcrafted, artisan build that never got properly replaced by the dynamodb serializer. 

"Version" could well be considered "optional" with newer code since we don't really care anymore about message order and inflight confusion.